rts/Game/GameHelper.cpp

```diff
@@ -1283,6 +1283,7 @@ CGameHelper::BuildSquareStatus CGameHelper::TestUnitBuildSquare(
 	// const float buildHeight = GetBuildHeight(testPos, buildInfo.def, synced);
 	// const float modelHeight = (model != nullptr) ? math::fabs(model->height) : 10.0f;
 
+	// FIXME this call is probably useless, we can assume that any caller already assigned it correctly
```

Comment:
can we? Lua can pass anything here.

Reply:
> The current interface in the PR is already good because it lets you specify position as XYZ. So it would be a matter of the renderer not ignoring the Y. Perhaps if that happens then Y could become optional so that specifying nil makes it use the unsynced ground height.

That was following following this comment, to read provided Y instead of recomputing it. I don't think we need to worry about LUA passing stupid values, they can call `GetBuildHeight` too. But that was following your comment, so your call to keep this FIXME around or not.

Reply:
I meant more in the wording sense. Games decide what is or isn't "correct", not engine, and it wasn't obvious what the practical effect of removal would be. I'd keep the line below it for now until it becomes possible to specify "use ground height" in addition to an explicit value for Y (because the ground height can change and it would otherwise be a pain to adjust existing squares) and just change the comment to be more explicit
```suggestion
	// FIXME remove this line to let Lua buildsquares specify arbitrary Y, if/when defaulting to ground height exists
```